migrations/1601022564938-copy-parents-data-into-children-entities-and-delete-parent-users.js

```diff
@@ -0,0 +1,166 @@
+				},
+			],
+		},
+	}),
+	'users'
+);
+const User = mongoose.model(
+	'myUserModel_05102020',
+	new mongoose.Schema({
+		roles: [{ type: mongoose.Schema.Types.ObjectId, ref: 'role' }],
+		firstName: { type: String, required: true },
+		lastName: { type: String, required: true },
+		email: { type: String, required: true, lowercase: true },
+		schoolId: {
+			type: mongoose.Schema.Types.ObjectId,
+			ref: 'school',
+			required: true,
+			index: true,
+		},
+		parents: [
+			{
+				firstName: { type: String, required: true },
+				lastName: { type: String, required: true },
+				email: { type: String, required: true, lowercase: true },
+			},
+		],
+		childrens: [{ type: mongoose.Schema.Types.ObjectId, ref: 'user' }],
+		consent: {
+			parentConsents: [
+				{
+					parentId: { type: mongoose.Schema.Types.ObjectId, ref: 'user' },
+					form: { type: String, enum: consentForm },
+					dateOfPrivacyConsent: { type: Date },
+					dateOfTermsOfUseConsent: { type: Date },
+					privacyConsent: { type: Boolean },
+					termsOfUseConsent: { type: Boolean },
+				},
+			],
+		},
+	}),
+	'users'
+);
+
+const getIds = (docs) => docs.map((doc) => doc._id);
+
+const getParentRole = async () => RoleModel.findOne({ name: 'parent' }).lean().exec();
+
+const findAllParents = (parentRole) => User.find({ roles: parentRole._id });
+
+const deleteParentUsersByIds = async (ids) => {
+	const result = await User.deleteMany({
+		_id: { $in: ids },
+	}).exec();
+	info(`${result.deletedCount} users deleted.`);
+};
+
+const getStudentsForParent = async (parent) => {
+	const students = await OldUser.find({ parents: parent._id }).lean().exec();
+	info(`${students.length} students for parent with id=${parent._id} will be updated!`);
+	return students;
+};
+
+const updateStudentsParentData = async (studentIds, parent) => {
+	return User.updateMany(
+		{ _id: { $in: [studentIds] } },
+		{
+			$set: {
+				parents: [
+					{
+						firstName: parent.firstName,
+						lastName: parent.lastName,
+						email: parent.email,
+					},
+				],
+			},
+			$unset: { 'consent.parentConsents.$[].parentId': '' },
+		}
+	);
+};
+
+const migrateParentsToStudents = async () => {
+	const parentIdsToDelete = [];
+	const parentRole = await getParentRole();
+	const cursor = findAllParents(parentRole).cursor();
+	for (let parent = await cursor.next(); parent != null; parent = await cursor.next()) {
+		const students = await getStudentsForParent(parent);
+		const studentIds = getIds(students);
+		await updateStudentsParentData(studentIds, parent);
+		parentIdsToDelete.push(parent._id);
+	}
+
+	let amountOfParentsLeftToDelete = parentIdsToDelete.length;
+
+	const limit = 500;
+	while (amountOfParentsLeftToDelete !== 0) {
+		const parentsToDelete = amountOfParentsLeftToDelete < limit ? amountOfParentsLeftToDelete : limit;
+		await deleteParentUsersByIds(parentIdsToDelete.splice(0, parentsToDelete));
+		amountOfParentsLeftToDelete -= parentsToDelete;
+	}
+};
```

Comment:
@Parnswir Shall we remove parent role at the end of migration?

Reply:
@Parnswir has approved, that parent role should be removed, already done